interop/test_utils.go

```diff
@@ -673,6 +676,89 @@ func DoPickFirstUnary(tc testgrpc.TestServiceClient) {
 	}
 }
 
+type soakIterationResult struct {
+	err     error
+	latency time.Duration
+}
+
+func doOneSoakIteration(tc testgrpc.TestServiceClient, resetChannel bool, serverAddr string, dopts []grpc.DialOption) error {
+	client := tc
+	if resetChannel {
+		conn, err := grpc.Dial(serverAddr, dopts...)
+		if err != nil {
+			return err
+		}
+		defer conn.Close()
+		client = testgrpc.NewTestServiceClient(conn)
+	}
+	// do a large-unary RPC
+	pl := ClientNewPayload(testpb.PayloadType_COMPRESSABLE, largeReqSize)
+	req := &testpb.SimpleRequest{
+		ResponseType: testpb.PayloadType_COMPRESSABLE,
+		ResponseSize: int32(largeRespSize),
+		Payload:      pl,
+	}
+	reply, err := client.UnaryCall(context.Background(), req)
```

Comment:
I understand the idea is to not set timeouts.

Should this set deadline to be the `overallDeadline`?
We want this test to fail at the time, right?

Reply:
good idea, done.

I figured the previous way would usually work well enough, unless the RPC got completely stuck. But your approach is ideal